ml_datasets/loaders/cifar.py

```diff
@@ -0,0 +1,81 @@
+import pickle
+import tarfile
+import random
+import numpy
+
+from ..util import get_file, unzip, to_categorical
+
+CIFAR10_URL = "https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz"
+CIFAR100_URL = "https://www.cs.toronto.edu/~kriz/cifar-100-python.tar.gz"
+
+
+def cifar(variant='10', channels_last=False, shuffle=True):
+    if variant == '10':
+        data = load_cifar10()
+    elif variant == '100':
+        data = load_cifar100()
+    elif variant == '100-coarse':
+        data = load_cifar100(coarse=True)
+    else:
+        raise ValueError("variant must be one of: '10', '100', 100-coarse")
+    X_train, y_train, X_test, y_test = data
+    X_train = X_train.astype("float32")
+    X_test = X_test.astype("float32")
+    X_train /= 255.0
+    X_test /= 255.0
+    if shuffle:
+        train_data = list(zip(X_train, y_train))
+        random.shuffle(train_data)
+        X_train, y_train = unzip(train_data)
+    if channels_last:
+        X_train = X_train.reshape(X_train.shape[0], 32, 32, 3)
+        X_test = X_test.reshape(X_test.shape[0], 32, 32, 3)
+    else:
+        X_train = X_train.reshape(X_train.shape[0], 3, 32, 32)
+        X_test = X_test.reshape(X_test.shape[0], 3, 32, 32)
+    y_train = to_categorical(y_train)
+    y_test = to_categorical(y_test)
+    return (X_train, y_train), (X_test, y_test)
+
+
+def load_cifar10(path='cifar-10-python.tar.gz'):
+    path = get_file(path, origin=CIFAR10_URL)
+    train_images = []
+    train_labels = []
+    with tarfile.open(path) as cifarf:
+        for name in cifarf.getnames():
+            # data is stored in batches
+            if 'data_batch' in name:
+                decompressed = cifarf.extractfile(name)
+                data = pickle.load(decompressed, encoding='bytes')
+                train_images.append(data[b'data'])
+                train_labels += data[b'labels']
+            elif 'test_batch' in name:
+                decompressed = cifarf.extractfile(name)
+                data = pickle.load(decompressed, encoding='bytes')
+                test_images = data[b'data']
+                test_labels = data[b'labels']
+    train_images = numpy.vstack(train_images)
+    train_labels = numpy.asarray(train_labels)
+    test_labels = numpy.asarray(train_labels)
```

Comment:
This would have made for interesting experiments 😛 
```suggestion
    test_labels = numpy.asarray(test_labels)
```

Reply:
oh no! Thank you!